src/main/java/it/smartcommunitylab/aac/oauth/auth/OAuth2IdpAwareLoginUrlConverter.java

```diff
@@ -80,7 +80,7 @@ public String convert(
                     throw new NoSuchProviderException();
                 }
 
-                return provider.getAuthenticationUrl();
+                return provider.getAuthenticationUrl(request);
```

Comment:
we can also improve the idp_hint logic and resolving providers based on their `name` and not only id: given the client, we can resolve the realm and then inside the realm pick the right provider by name.

This would let clients build requests as 
```
authorize?idp_hint=spid&domain_hint...
```